File: binding/include/network/libqsslconfiguration.zig
const QtC = @import("qt6zig");
const qtc = @import("qt6c");%%_IMPORTLIBS_%%
%%_STRUCTDEFS_%%

/// https://doc.qt.io/qt-6/qsslconfiguration.html
pub const qsslconfiguration = struct {

    /// New constructs a new QSslConfiguration object.
    pub fn New() QtC.QSslConfiguration {
        return qtc.QSslConfiguration_new();
    }


    /// New2 constructs a new QSslConfiguration object.
    pub fn New2(other: ?*anyopaque) QtC.QSslConfiguration {
        return qtc.QSslConfiguration_new2(@ptrCast(other));
    }


    pub fn OperatorAssign(self: ?*anyopaque, other: ?*anyopaque) void {
        qtc.QSslConfiguration_OperatorAssign(@ptrCast(self), @ptrCast(other));
    }

    pub fn Swap(self: ?*anyopaque, other: ?*anyopaque) void {
        qtc.QSslConfiguration_Swap(@ptrCast(self), @ptrCast(other));
    }

    pub fn OperatorEqual(self: ?*anyopaque, other: ?*anyopaque) bool {
        return qtc.QSslConfiguration_OperatorEqual(@ptrCast(self), @ptrCast(other));
    }

    pub fn OperatorNotEqual(self: ?*anyopaque, other: ?*anyopaque) bool {
        return qtc.QSslConfiguration_OperatorNotEqual(@ptrCast(self), @ptrCast(other));
    }

    pub fn IsNull(self: ?*anyopaque, ) bool {
        return qtc.QSslConfiguration_IsNull(@ptrCast(self));
    }

    pub fn Protocol(self: ?*anyopaque, ) i32 {
        return qtc.QSslConfiguration_Protocol(@ptrCast(self));
    }

    pub fn SetProtocol(self: ?*anyopaque, protocol: i32) void {
        qtc.QSslConfiguration_SetProtocol(@ptrCast(self), @intCast(protocol));
    }

    pub fn PeerVerifyMode(self: ?*anyopaque, ) i32 {
        return qtc.QSslConfiguration_PeerVerifyMode(@ptrCast(self));
    }

    pub fn SetPeerVerifyMode(self: ?*anyopaque, mode: i32) void {
        qtc.QSslConfiguration_SetPeerVerifyMode(@ptrCast(self), @intCast(mode));
    }

    pub fn PeerVerifyDepth(self: ?*anyopaque, ) i32 {
        return qtc.QSslConfiguration_PeerVerifyDepth(@ptrCast(self));
    }

    pub fn SetPeerVerifyDepth(self: ?*anyopaque, depth: i32) void {
        qtc.QSslConfiguration_SetPeerVerifyDepth(@ptrCast(self), depth);
    }

    pub fn LocalCertificateChain(self: ?*anyopaque, ) []const u8 {
        return qtc.QSslConfiguration_LocalCertificateChain(@ptrCast(self));
    }

    pub fn SetLocalCertificateChain(self: ?*anyopaque, localChain: []const u8) void {
        qtc.QSslConfiguration_SetLocalCertificateChain(@ptrCast(self), @ptrCast(localChain));
    }

    pub fn LocalCertificate(self: ?*anyopaque, ) QtC.QSslCertificate {
        return qtc.QSslConfiguration_LocalCertificate(@ptrCast(self));
    }

    pub fn SetLocalCertificate(self: ?*anyopaque, certificate: ?*anyopaque) void {
        qtc.QSslConfiguration_SetLocalCertificate(@ptrCast(self), @ptrCast(certificate));
    }

    pub fn PeerCertificate(self: ?*anyopaque, ) QtC.QSslCertificate {
        return qtc.QSslConfiguration_PeerCertificate(@ptrCast(self));
    }

    pub fn PeerCertificateChain(self: ?*anyopaque, ) []const u8 {
        return qtc.QSslConfiguration_PeerCertificateChain(@ptrCast(self));
    }

    pub fn SessionCipher(self: ?*anyopaque, ) QtC.QSslCipher {
        return qtc.QSslConfiguration_SessionCipher(@ptrCast(self));
    }

    pub fn SessionProtocol(self: ?*anyopaque, ) i32 {
        return qtc.QSslConfiguration_SessionProtocol(@ptrCast(self));
    }

    pub fn PrivateKey(self: ?*anyopaque, ) QtC.QSslKey {
        return qtc.QSslConfiguration_PrivateKey(@ptrCast(self));
    }

    pub fn SetPrivateKey(self: ?*anyopaque, key: ?*anyopaque) void {
        qtc.QSslConfiguration_SetPrivateKey(@ptrCast(self), @ptrCast(key));
    }

    pub fn Ciphers(self: ?*anyopaque, ) []const u8 {
        return qtc.QSslConfiguration_Ciphers(@ptrCast(self));
    }

    pub fn SetCiphers(self: ?*anyopaque, ciphers: []const u8) void {
        qtc.QSslConfiguration_SetCiphers(@ptrCast(self), @ptrCast(ciphers));
    }

    pub fn SetCiphers2(self: ?*anyopaque, ciphers: []const u8) void {
        const ciphers_str = qtc.libqt_string{
    .len = ciphers.len,
    .data = ciphers.ptr,
};
qtc.QSslConfiguration_SetCiphers2(@ptrCast(self), ciphers_str);
    }

    pub fn SupportedCiphers() []const u8 {
        return qtc.QSslConfiguration_SupportedCiphers();
    }

    pub fn CaCertificates(self: ?*anyopaque, ) []const u8 {
        return qtc.QSslConfiguration_CaCertificates(@ptrCast(self));
    }

    pub fn SetCaCertificates(self: ?*anyopaque, certificates: []const u8) void {
        qtc.QSslConfiguration_SetCaCertificates(@ptrCast(self), @ptrCast(certificates));
    }

    pub fn AddCaCertificates(self: ?*anyopaque, path: []const u8) bool {
        const path_str = qtc.libqt_string{
    .len = path.len,
    .data = path.ptr,
};
return qtc.QSslConfiguration_AddCaCertificates(@ptrCast(self), path_str);
    }

    pub fn AddCaCertificate(self: ?*anyopaque, certificate: ?*anyopaque) void {
        qtc.QSslConfiguration_AddCaCertificate(@ptrCast(self), @ptrCast(certificate));
    }

    pub fn AddCaCertificates2(self: ?*anyopaque, certificates: []const u8) void {
        qtc.QSslConfiguration_AddCaCertificates2(@ptrCast(self), @ptrCast(certificates));
    }

    pub fn SystemCaCertificates() []const u8 {
        return qtc.QSslConfiguration_SystemCaCertificates();
    }

    pub fn SetSslOption(self: ?*anyopaque, option: i32, on: bool) void {
        qtc.QSslConfiguration_SetSslOption(@ptrCast(self), @intCast(option), on);
    }

    pub fn TestSslOption(self: ?*anyopaque, option: i32) bool {
        return qtc.QSslConfiguration_TestSslOption(@ptrCast(self), @intCast(option));
    }

    pub fn SessionTicket(self: ?*anyopaque, , allocator: std.mem.Allocator) []u8 {
        const _bytearray: qtc.libqt_string = qtc.QSslConfiguration_SessionTicket(@ptrCast(self));
defer qtc.libqt_string_free(&_bytearray);
const _ret = allocator.alloc(u8, _bytearray.len) catch @panic("qsslconfiguration.SessionTicket: Memory allocation failed");
@memcpy(_ret, _bytearray.data[0.._bytearray.len]);
return  _ret;
    }

    pub fn SetSessionTicket(self: ?*anyopaque, sessionTicket: []u8) void {
        const sessionTicket_str = qtc.libqt_string{
    .len = sessionTicket.len,
    .data = sessionTicket.ptr,
};
qtc.QSslConfiguration_SetSessionTicket(@ptrCast(self), sessionTicket_str);
    }

    pub fn SessionTicketLifeTimeHint(self: ?*anyopaque, ) i32 {
        return qtc.QSslConfiguration_SessionTicketLifeTimeHint(@ptrCast(self));
    }

    pub fn EphemeralServerKey(self: ?*anyopaque, ) QtC.QSslKey {
        return qtc.QSslConfiguration_EphemeralServerKey(@ptrCast(self));
    }

    pub fn EllipticCurves(self: ?*anyopaque, ) []const u8 {
        return qtc.QSslConfiguration_EllipticCurves(@ptrCast(self));
    }

    pub fn SetEllipticCurves(self: ?*anyopaque, curves: []const u8) void {
        qtc.QSslConfiguration_SetEllipticCurves(@ptrCast(self), @ptrCast(curves));
    }

    pub fn SupportedEllipticCurves() []const u8 {
        return qtc.QSslConfiguration_SupportedEllipticCurves();
    }

    pub fn PreSharedKeyIdentityHint(self: ?*anyopaque, , allocator: std.mem.Allocator) []u8 {
        const _bytearray: qtc.libqt_string = qtc.QSslConfiguration_PreSharedKeyIdentityHint(@ptrCast(self));
defer qtc.libqt_string_free(&_bytearray);
const _ret = allocator.alloc(u8, _bytearray.len) catch @panic("qsslconfiguration.PreSharedKeyIdentityHint: Memory allocation failed");
@memcpy(_ret, _bytearray.data[0.._bytearray.len]);
return  _ret;
    }

    pub fn SetPreSharedKeyIdentityHint(self: ?*anyopaque, hint: []u8) void {
        const hint_str = qtc.libqt_string{
    .len = hint.len,
    .data = hint.ptr,
};
qtc.QSslConfiguration_SetPreSharedKeyIdentityHint(@ptrCast(self), hint_str);
    }

    pub fn DiffieHellmanParameters(self: ?*anyopaque, ) QtC.QSslDiffieHellmanParameters {
        return qtc.QSslConfiguration_DiffieHellmanParameters(@ptrCast(self));
    }

    pub fn SetDiffieHellmanParameters(self: ?*anyopaque, dhparams: ?*anyopaque) void {
        qtc.QSslConfiguration_SetDiffieHellmanParameters(@ptrCast(self), @ptrCast(dhparams));
    }

    pub fn BackendConfiguration(self: ?*anyopaque, ) QMap<QByteArray, QVariant> {
        return qtc.QSslConfiguration_BackendConfiguration(@ptrCast(self));
    }

    pub fn SetBackendConfigurationOption(self: ?*anyopaque, name: []u8, value: ?*anyopaque) void {
        const name_str = qtc.libqt_string{
    .len = name.len,
    .data = name.ptr,
};
qtc.QSslConfiguration_SetBackendConfigurationOption(@ptrCast(self), name_str, @ptrCast(value));
    }

    pub fn SetBackendConfiguration(self: ?*anyopaque, ) void {
        qtc.QSslConfiguration_SetBackendConfiguration(@ptrCast(self));
    }

    pub fn DefaultConfiguration() QtC.QSslConfiguration {
        return qtc.QSslConfiguration_DefaultConfiguration();
    }

    pub fn SetDefaultConfiguration(configuration: ?*anyopaque) void {
        qtc.QSslConfiguration_SetDefaultConfiguration(@ptrCast(configuration));
    }

    pub fn DtlsCookieVerificationEnabled(self: ?*anyopaque, ) bool {
        return qtc.QSslConfiguration_DtlsCookieVerificationEnabled(@ptrCast(self));
    }

    pub fn SetDtlsCookieVerificationEnabled(self: ?*anyopaque, enable: bool) void {
        qtc.QSslConfiguration_SetDtlsCookieVerificationEnabled(@ptrCast(self), enable);
    }

    pub fn DefaultDtlsConfiguration() QtC.QSslConfiguration {
        return qtc.QSslConfiguration_DefaultDtlsConfiguration();
    }

    pub fn SetDefaultDtlsConfiguration(configuration: ?*anyopaque) void {
        qtc.QSslConfiguration_SetDefaultDtlsConfiguration(@ptrCast(configuration));
    }

    pub fn HandshakeMustInterruptOnError(self: ?*anyopaque, ) bool {
        return qtc.QSslConfiguration_HandshakeMustInterruptOnError(@ptrCast(self));
    }

    pub fn SetHandshakeMustInterruptOnError(self: ?*anyopaque, interrupt: bool) void {
        qtc.QSslConfiguration_SetHandshakeMustInterruptOnError(@ptrCast(self), interrupt);
    }

    pub fn MissingCertificateIsFatal(self: ?*anyopaque, ) bool {
        return qtc.QSslConfiguration_MissingCertificateIsFatal(@ptrCast(self));
    }

    pub fn SetMissingCertificateIsFatal(self: ?*anyopaque, cannotRecover: bool) void {
        qtc.QSslConfiguration_SetMissingCertificateIsFatal(@ptrCast(self), cannotRecover);
    }

    pub fn SetOcspStaplingEnabled(self: ?*anyopaque, enable: bool) void {
        qtc.QSslConfiguration_SetOcspStaplingEnabled(@ptrCast(self), enable);
    }

    pub fn OcspStaplingEnabled(self: ?*anyopaque, ) bool {
        return qtc.QSslConfiguration_OcspStaplingEnabled(@ptrCast(self));
    }

    pub fn SetAllowedNextProtocols(self: ?*anyopaque, protocols: []const u8) void {
        qtc.QSslConfiguration_SetAllowedNextProtocols(@ptrCast(self), @ptrCast(protocols));
    }

    pub fn AllowedNextProtocols(self: ?*anyopaque, ) []const u8 {
        return qtc.QSslConfiguration_AllowedNextProtocols(@ptrCast(self));
    }

    pub fn NextNegotiatedProtocol(self: ?*anyopaque, , allocator: std.mem.Allocator) []u8 {
        const _bytearray: qtc.libqt_string = qtc.QSslConfiguration_NextNegotiatedProtocol(@ptrCast(self));
defer qtc.libqt_string_free(&_bytearray);
const _ret = allocator.alloc(u8, _bytearray.len) catch @panic("qsslconfiguration.NextNegotiatedProtocol: Memory allocation failed");
@memcpy(_ret, _bytearray.data[0.._bytearray.len]);
return  _ret;
    }

    pub fn NextProtocolNegotiationStatus(self: ?*anyopaque, ) i32 {
        return qtc.QSslConfiguration_NextProtocolNegotiationStatus(@ptrCast(self));
    }

    pub fn AddCaCertificates22(self: ?*anyopaque, path: []const u8, format: i32) bool {
        const path_str = qtc.libqt_string{
    .len = path.len,
    .data = path.ptr,
};
return qtc.QSslConfiguration_AddCaCertificates22(@ptrCast(self), path_str, @intCast(format));
    }

    pub fn AddCaCertificates3(self: ?*anyopaque, path: []const u8, format: i32, syntax: i32) bool {
        const path_str = qtc.libqt_string{
    .len = path.len,
    .data = path.ptr,
};
return qtc.QSslConfiguration_AddCaCertificates3(@ptrCast(self), path_str, @intCast(format), @intCast(syntax));
    }

    pub fn SetBackendConfiguration1(self: ?*anyopaque, backendConfiguration: *const QMap<QByteArray, QVariant>) void {
        qtc.QSslConfiguration_SetBackendConfiguration1(@ptrCast(self), @ptrCast(backendConfiguration));
    }

    /// Delete this object from C++ memory.
    pub fn QDelete(self: ?*anyopaque) void {
        qtc.QSslConfiguration_Delete(@ptrCast(self));
    }
};

/// https://doc.qt.io/qt-6/sslconfiguration.html#types
pub const enums = struct {
    pub const NextProtocolNegotiationStatus = enum {
        pub const NextProtocolNegotiationNone: i32 = 0;
        pub const NextProtocolNegotiationNegotiated: i32 = 1;
        pub const NextProtocolNegotiationUnsupported: i32 = 2;
    };

};

File: binding/src/network/libqsslconfiguration.cpp
#include <QByteArray>
#include <QSslCertificate>
#include <QSslCipher>
#include <QSslConfiguration>
#include <QSslDiffieHellmanParameters>
#include <QSslKey>
#include <QString>
#include <QByteArray>
#include <cstring>
#include <QVariant>
#include <qsslconfiguration.h>
#include "libqsslconfiguration.h"
#include "libqsslconfiguration.hxx"

QSslConfiguration* QSslConfiguration_new() {
	 return new QSslConfiguration();
}

QSslConfiguration* QSslConfiguration_new2(const QSslConfiguration* other) {
	 return new QSslConfiguration(*other);
}

void QSslConfiguration_OperatorAssign(QSslConfiguration* self, const QSslConfiguration* other) {
	self->operator=(*other);
}

void QSslConfiguration_Swap(QSslConfiguration* self, QSslConfiguration* other) {
	self->swap(*other);
}

bool QSslConfiguration_OperatorEqual(const QSslConfiguration* self, const QSslConfiguration* other) {
	return self->operator==(*other);
}

bool QSslConfiguration_OperatorNotEqual(const QSslConfiguration* self, const QSslConfiguration* other) {
	return self->operator!=(*other);
}

bool QSslConfiguration_IsNull(const QSslConfiguration* self) {
	return self->isNull();
}

int QSslConfiguration_Protocol(const QSslConfiguration* self) {
	return self->protocol();
}

void QSslConfiguration_SetProtocol(QSslConfiguration* self, int protocol) {
	self->setProtocol(static_cast<QSsl::SslProtocol>(protocol));
}

int QSslConfiguration_PeerVerifyMode(const QSslConfiguration* self) {
	return self->peerVerifyMode();
}

void QSslConfiguration_SetPeerVerifyMode(QSslConfiguration* self, int mode) {
	self->setPeerVerifyMode(static_cast<QSslSocket::PeerVerifyMode>(mode));
}

int QSslConfiguration_PeerVerifyDepth(const QSslConfiguration* self) {
	return self->peerVerifyDepth();
}

void QSslConfiguration_SetPeerVerifyDepth(QSslConfiguration* self, int depth) {
	self->setPeerVerifyDepth(depth);
}

libqt_list QSslConfiguration_LocalCertificateChain(const QSslConfiguration* self) {
	return self->localCertificateChain();
}

void QSslConfiguration_SetLocalCertificateChain(QSslConfiguration* self, const libqt_list localChain) {
	self->setLocalCertificateChain(*localChain);
}

QSslCertificate* QSslConfiguration_LocalCertificate(const QSslConfiguration* self) {
	return new QSslCertificate(self->localCertificate());
}

void QSslConfiguration_SetLocalCertificate(QSslConfiguration* self, const QSslCertificate* certificate) {
	self->setLocalCertificate(*certificate);
}

QSslCertificate* QSslConfiguration_PeerCertificate(const QSslConfiguration* self) {
	return new QSslCertificate(self->peerCertificate());
}

libqt_list QSslConfiguration_PeerCertificateChain(const QSslConfiguration* self) {
	return self->peerCertificateChain();
}

QSslCipher* QSslConfiguration_SessionCipher(const QSslConfiguration* self) {
	return new QSslCipher(self->sessionCipher());
}

int QSslConfiguration_SessionProtocol(const QSslConfiguration* self) {
	return self->sessionProtocol();
}

QSslKey* QSslConfiguration_PrivateKey(const QSslConfiguration* self) {
	return new QSslKey(self->privateKey());
}

void QSslConfiguration_SetPrivateKey(QSslConfiguration* self, const QSslKey* key) {
	self->setPrivateKey(*key);
}

libqt_list QSslConfiguration_Ciphers(const QSslConfiguration* self) {
	return self->ciphers();
}

void QSslConfiguration_SetCiphers(QSslConfiguration* self, const libqt_list ciphers) {
	self->setCiphers(*ciphers);
}

void QSslConfiguration_SetCiphers2(QSslConfiguration* self, const libqt_string ciphers) {
	self->setCiphers(QString::fromUtf8(ciphers.data, ciphers.len));
}

libqt_list QSslConfiguration_SupportedCiphers() {
	return QSslConfiguration::supportedCiphers();
}

libqt_list QSslConfiguration_CaCertificates(const QSslConfiguration* self) {
	return self->caCertificates();
}

void QSslConfiguration_SetCaCertificates(QSslConfiguration* self, const libqt_list certificates) {
	self->setCaCertificates(*certificates);
}

bool QSslConfiguration_AddCaCertificates(QSslConfiguration* self, const libqt_string path) {
	return self->addCaCertificates(QString::fromUtf8(path.data, path.len));
}

void QSslConfiguration_AddCaCertificate(QSslConfiguration* self, const QSslCertificate* certificate) {
	self->addCaCertificate(*certificate);
}

void QSslConfiguration_AddCaCertificates2(QSslConfiguration* self, const libqt_list certificates) {
	self->addCaCertificates(*certificates);
}

libqt_list QSslConfiguration_SystemCaCertificates() {
	return QSslConfiguration::systemCaCertificates();
}

void QSslConfiguration_SetSslOption(QSslConfiguration* self, int option, bool on) {
	self->setSslOption(static_cast<QSsl::SslOption>(option), on);
}

bool QSslConfiguration_TestSslOption(const QSslConfiguration* self, int option) {
	return self->testSslOption(static_cast<QSsl::SslOption>(option));
}

libqt_string QSslConfiguration_SessionTicket(const QSslConfiguration* self) {
	QByteArray _qb = self->sessionTicket();
	libqt_string _str;
	_str.len = _qb.length();
	_str.data = static_cast<const char*>(malloc(_str.len + 1));
	memcpy((void*)_str.data, _qb.data(), _str.len);
	((char*)_str.data)[_str.len] = '\0';
	return _str;
}

void QSslConfiguration_SetSessionTicket(QSslConfiguration* self, const libqt_string sessionTicket) {
	self->setSessionTicket(QByteArray(sessionTicket.data, sessionTicket.len));
}

int QSslConfiguration_SessionTicketLifeTimeHint(const QSslConfiguration* self) {
	return self->sessionTicketLifeTimeHint();
}

QSslKey* QSslConfiguration_EphemeralServerKey(const QSslConfiguration* self) {
	return new QSslKey(self->ephemeralServerKey());
}

libqt_list QSslConfiguration_EllipticCurves(const QSslConfiguration* self) {
	return self->ellipticCurves();
}

void QSslConfiguration_SetEllipticCurves(QSslConfiguration* self, const libqt_list curves) {
	self->setEllipticCurves(*curves);
}

libqt_list QSslConfiguration_SupportedEllipticCurves() {
	return QSslConfiguration::supportedEllipticCurves();
}

libqt_string QSslConfiguration_PreSharedKeyIdentityHint(const QSslConfiguration* self) {
	QByteArray _qb = self->preSharedKeyIdentityHint();
	libqt_string _str;
	_str.len = _qb.length();
	_str.data = static_cast<const char*>(malloc(_str.len + 1));
	memcpy((void*)_str.data, _qb.data(), _str.len);
	((char*)_str.data)[_str.len] = '\0';
	return _str;
}

void QSslConfiguration_SetPreSharedKeyIdentityHint(QSslConfiguration* self, const libqt_string hint) {
	self->setPreSharedKeyIdentityHint(QByteArray(hint.data, hint.len));
}

QSslDiffieHellmanParameters* QSslConfiguration_DiffieHellmanParameters(const QSslConfiguration* self) {
	return new QSslDiffieHellmanParameters(self->diffieHellmanParameters());
}

void QSslConfiguration_SetDiffieHellmanParameters(QSslConfiguration* self, const QSslDiffieHellmanParameters* dhparams) {
	self->setDiffieHellmanParameters(*dhparams);
}

libqt_map QSslConfiguration_BackendConfiguration(const QSslConfiguration* self) {
	return self->backendConfiguration();
}

void QSslConfiguration_SetBackendConfigurationOption(QSslConfiguration* self, const libqt_string name, const QVariant* value) {
	self->setBackendConfigurationOption(QByteArray(name.data, name.len), *value);
}

void QSslConfiguration_SetBackendConfiguration(QSslConfiguration* self) {
	self->setBackendConfiguration();
}

QSslConfiguration* QSslConfiguration_DefaultConfiguration() {
	return new QSslConfiguration(QSslConfiguration::defaultConfiguration());
}

void QSslConfiguration_SetDefaultConfiguration(const QSslConfiguration* configuration) {
	QSslConfiguration::setDefaultConfiguration(*configuration);
}

bool QSslConfiguration_DtlsCookieVerificationEnabled(const QSslConfiguration* self) {
	return self->dtlsCookieVerificationEnabled();
}

void QSslConfiguration_SetDtlsCookieVerificationEnabled(QSslConfiguration* self, bool enable) {
	self->setDtlsCookieVerificationEnabled(enable);
}

QSslConfiguration* QSslConfiguration_DefaultDtlsConfiguration() {
	return new QSslConfiguration(QSslConfiguration::defaultDtlsConfiguration());
}

void QSslConfiguration_SetDefaultDtlsConfiguration(const QSslConfiguration* configuration) {
	QSslConfiguration::setDefaultDtlsConfiguration(*configuration);
}

bool QSslConfiguration_HandshakeMustInterruptOnError(const QSslConfiguration* self) {
	return self->handshakeMustInterruptOnError();
}

void QSslConfiguration_SetHandshakeMustInterruptOnError(QSslConfiguration* self, bool interrupt) {
	self->setHandshakeMustInterruptOnError(interrupt);
}

bool QSslConfiguration_MissingCertificateIsFatal(const QSslConfiguration* self) {
	return self->missingCertificateIsFatal();
}

void QSslConfiguration_SetMissingCertificateIsFatal(QSslConfiguration* self, bool cannotRecover) {
	self->setMissingCertificateIsFatal(cannotRecover);
}

void QSslConfiguration_SetOcspStaplingEnabled(QSslConfiguration* self, bool enable) {
	self->setOcspStaplingEnabled(enable);
}

bool QSslConfiguration_OcspStaplingEnabled(const QSslConfiguration* self) {
	return self->ocspStaplingEnabled();
}

void QSslConfiguration_SetAllowedNextProtocols(QSslConfiguration* self, const libqt_list protocols) {
	self->setAllowedNextProtocols(*protocols);
}

libqt_list QSslConfiguration_AllowedNextProtocols(const QSslConfiguration* self) {
	return self->allowedNextProtocols();
}

libqt_string QSslConfiguration_NextNegotiatedProtocol(const QSslConfiguration* self) {
	QByteArray _qb = self->nextNegotiatedProtocol();
	libqt_string _str;
	_str.len = _qb.length();
	_str.data = static_cast<const char*>(malloc(_str.len + 1));
	memcpy((void*)_str.data, _qb.data(), _str.len);
	((char*)_str.data)[_str.len] = '\0';
	return _str;
}

int QSslConfiguration_NextProtocolNegotiationStatus(const QSslConfiguration* self) {
	return self->nextProtocolNegotiationStatus();
}

bool QSslConfiguration_AddCaCertificates22(QSslConfiguration* self, const libqt_string path, int format) {
	return self->addCaCertificates(QString::fromUtf8(path.data, path.len), static_cast<QSsl::EncodingFormat>(format));
}

bool QSslConfiguration_AddCaCertificates3(QSslConfiguration* self, const libqt_string path, int format, int syntax) {
	return self->addCaCertificates(QString::fromUtf8(path.data, path.len), static_cast<QSsl::EncodingFormat>(format), static_cast<QSslCertificate::PatternSyntax>(syntax));
}

void QSslConfiguration_SetBackendConfiguration1(QSslConfiguration* self, const libqt_map backendConfiguration) {
	self->setBackendConfiguration(*backendConfiguration);
}

void QSslConfiguration_Delete(QSslConfiguration* self) {
    delete self;
}

